cosmetics-web/app/forms/notification_wizard/delete_nano_material_form.rb

```diff
@@ -0,0 +1,25 @@
+module NotificationWizard
+  class DeleteNanoMaterialForm < Form
+    class NonEmptyArrayValidator < ActiveModel::EachValidator
+      def validate_each(record, attribute, value)
+        if value.blank? || value.reject(&:blank?).blank?
+          record.errors.add attribute, "Please select nanomaterial to remove"
+        end
+      end
+    end
+
+    attribute :nano_material_ids
+    attribute :notification
+
+    validates :nano_material_ids, non_empty_array: true
+
+    def delete
+      return false unless valid?
+      raise ActiveRecord::RecordNotFound if %w[notification_complete deleted].include?(notification.state)
```

Comment:
Suggestion: replace hardcoded states for constants or the state helper methods:
```suggestion
      raise ActiveRecord::RecordNotFound if notification.notification_complete? || notification.deleted?
```